cloud/events_log.go

```diff
@@ -0,0 +1,113 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+type AuditService service
+
+type CustomTime struct {
+	time.Time
+}
+
+func (ct *CustomTime) UnmarshalJSON(b []byte) error {
+	s := string(b)
+	if s == "null" {
+		return nil
+	}
+	s = s[1 : len(s)-1]
+	layout := "2006-01-02T15:04:05.000-0700"
+	t, err := time.Parse(layout, s)
+	if err != nil {
+		return fmt.Errorf("CustomTime unmarshal error: %w", err)
+	}
+	ct.Time = t
+	return nil
+}
+
+type AuditRecord struct {
+	ID              int64                 `json:"id"`
+	Summary         string                `json:"summary"`
+	Created         CustomTime            `json:"created"`
+	Category        string                `json:"category"`
+	EventSource     string                `json:"eventSource"`
+	ObjectItem      AuditObjectItem       `json:"objectItem"`
+	ChangedValues   []AuditChangedValue   `json:"changedValues"`
+	AssociatedItems []AuditAssociatedItem `json:"associatedItems"`
+	RemoteAddress   string                `json:"remoteAddress"`
+	AuthorKey       string                `json:"authorKey"`
+	AuthorAccountId string                `json:"authorAccountId"`
+	Description     string                `json:"description"`
+}
+
+type AuditObjectItem struct {
+	ID         string `json:"id"`
+	Name       string `json:"name"`
+	TypeName   string `json:"typeName"`
+	ParentId   string `json:"parentId"`
+	ParentName string `json:"parentName"`
+}
+
+type AuditChangedValue struct {
+	FieldName   string `json:"fieldName"`
+	ChangedFrom string `json:"changedFrom"`
+	ChangedTo   string `json:"changedTo"`
+}
+
+type AuditAssociatedItem struct {
+	ID         string `json:"id"`
+	Name       string `json:"name"`
+	TypeName   string `json:"typeName"`
+	ParentId   string `json:"parentId"`
+	ParentName string `json:"parentName"`
+}
+
+type AuditResponse struct {
+	Offset  int64         `json:"offset"`
+	Limit   int64         `json:"limit"`
+	Total   int64         `json:"total"`
+	Records []AuditRecord `json:"records"`
+}
+
+func (s *AuditService) Get(ctx context.Context, from time.Time, offset, limit int) (*AuditResponse, *Response, error) {
```

Comment:
nit: we usually specify each type for each parameter art c1  `offset int, limit int` instead of `offset, limit int`